pandas/tests/io/test_user_agent.py

```diff
@@ -0,0 +1,309 @@
+"""
+Tests for the pandas custom headers in http(s) requests
+"""
+import gzip
+import http.server
+from io import BytesIO
+import threading
+
+import pytest
+
+import pandas as pd
+import pandas._testing as tm
+
```

Comment:
I think the confounding thing here is that not having fsspec installed breaks a test, and not the actual feature being tested (pandas could set custom http headers on a FastParquet file request and parse it without fsspec being installed).  The fsspec package is only needed in the case of testing FastParquet because it won't write to a BytesIO or filelike object without being patched.  I didn't importorskip in the class definition because it's outside of a test, and used inside a test.  I don't know if that helped explain it or made it less clear.  If you'd rather it be done another way, that's all good just let me know!